### PR TITLE
Adding rspecs for the API Client

### PR DESCRIPTION
**Pull request by abellotti**
Adding rspecs for the API Client

- [x] Actions
- [x] Action Results
- [x] Api
- [x]  Api Version
- [x] Authentication
- [x] Client
[x] Collection
- [x] Collection Options
[x] Connection
- [x] Error
- [x] Identity
- [x] Logger
- [x] Product Info
[x] Resource
- [x] Server Info
- [x] Version

### CODE REVIEW

File: spec/manageiq/api/action_result_spec.rb [hunk 1]
```diff
@@ -0,0 +1,76 @@
+describe ManageIQ::API::Client::ActionResult do
+  let(:invalid_action_result_error) { "Not a valid Action Result specified" }
+  let(:success_action_result) { {"success" => true, "message" => "Vm powered on"} }
+  let(:failed_action_result) { {"success" => false, "message" => "Vm could not be powered on"} }
+
+  describe "#new" do
```
Comment (Fryguy) [inline spec/manageiq/api/action_result_spec.rb]: Minor but this is a class method and this should be described as `".new"`

Comment (Fryguy) [inline spec/manageiq/api/action_result_spec.rb]: Same goes for all of the other class methods below. `#` indicates an instance method.

**Comment by Fryguy** [comment]
Can we merge these now and get the rest in a follow up PR?

**Comment by abellotti** [comment]
sure, they are in a good shape.  I'll unwip.

**Comment by b-long** [comment]
Any update on this?

**Comment by Fryguy** [comment]
Thanks for the bump @b-long .  I hadn't seen @abellotti's response :)

**Pull request merged**